eclipse-collections/src/main/java/com/fasterxml/jackson/datatype/eclipsecollections/deser/map/EclipseMapDeserializers.java

```diff
@@ -271,13 +275,28 @@ private static final class Entry<T, I, K extends KeyHandler<K>, V extends ValueH
             this.finish = finish;
         }
 
-        EclipseMapDeserializer<T, I, K, V> createDeserializer(JavaType type) {
+        EclipseMapDeserializer<T, I, K, V> createDeserializer(JavaType type) throws JsonMappingException {
             Class<?> rawClass = type.getRawClass();
             List<JavaType> typeParameters = type.getBindings().getTypeParameters();
             boolean refValue = PrimitiveObjectMap.class.isAssignableFrom(rawClass) ||
                                MapIterable.class.isAssignableFrom(rawClass);
             boolean refKey = refValue ? (typeParameters.size() == 2) : (typeParameters.size() == 1);
 
+            // Generic key type validation
+            if (typeHandlerPair == TypeHandlerPair.COMPARABLE_OBJECT) {
+                Class<?> expectedKeyClass = Comparable.class;
+                Class<?> actualKeyClass = typeParameters.get(0).getRawClass();
+
+                if (!expectedKeyClass.isAssignableFrom(actualKeyClass)) {
+                    String message = String.format(
+                        "Cannot deserialize %s: key type %s is not assignable to required type %s",
+                        rawClass.getSimpleName(),
+                        actualKeyClass.getName(),
+                        expectedKeyClass.getName()
+                    );
+                    throw new JsonMappingException(null, message);
```

Comment:
Was hoping to find a more specific databind exception, but there isn't one we can use (`InvalidDefinitionException` is almost it but... we don't have information to properly construct one).
So we'll go with base `JsonMappingException`.